app/Http/Requests/UpdateProductRequest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Models\Product;
+use Illuminate\Foundation\Http\FormRequest;
+
+class UpdateProductRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $productId = Product::where('slug', $this->slug)->first()->id;
```

Comment:
Tạm thời đừng dùng đến `slug`
Hãy dùng `id` để quen với cách CRUD cơ bản, sau này áp dụng được với các bài toán khác tốt hơn.

Reply:
Vâng, tại hôm a bảo k nên truyền id nên e mới tham khảo các anh bảo truyền slug vào